impl/test/indexrepotest.go

```diff
@@ -0,0 +1,57 @@
+package test
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/filecoin-project/dagstore/index"
+	"github.com/filecoin-project/dagstore/shard"
+	"github.com/ipfs/go-cid"
+	"github.com/stretchr/testify/require"
+)
+
+func RunIndexRepoTest(t *testing.T, r index.FullIndexRepo) {
```

Comment:
It's possible that we will have multiple implementations of the index repo going forward (such as when we move to run a distributed DAG store). Right now we have a "forced" one (memory repo), which we'll never use in production, but it's useful for tests. Instead of having test indirection, I would prefer to use subtests.